pkg/build/builder/cmd/builder.go

```diff
@@ -35,6 +36,7 @@ type builderConfig struct {
 	dockerClient    *docker.Client
 	dockerEndpoint  string
 	buildsClient    client.BuildInterface
+	eventsClient    kclient.EventInterface
```

Comment:
I discussed this with @derekwaynecarr. We don't need to be absolutely reliable. If the event doesn't get propagated or isn't catched, user will be still greeted with `denied error` which can happen only in this case.

Another option is to store the errors in image streams but I'd rather avoid it. With the quota we're protecting storage and etcd against the abuse. So if someone tries to abuse us with tagging a lot of images, storing all the failed attempts in etcd in preserved object will let him abuse us easily.